modules/contacts/client/components/ContactsCommon.component.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+// import { NamespacesConsumer } from 'react-i18next';
+import PropTypes from 'prop-types';
+import Contact from './Contact.component';
+import { contactsCommonListService } from '../api/contacts.api';
+import '@/config/lib/i18n';
+
+export default class ContactsCommon extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      contacts: []
+    };
+    this.isContactsCommonList = this.isContactsCommonList.bind(this);
+    this.onChange = this.onChange.bind(this);
+  }
+  isContactsCommonList() {
```

Comment:
The intention of this function is great: return no element when we have no contacts in common. However the return is applied to this function, not to the render(), and the full (empty) contact list is still displayed.

Did I understand the intention correctly?

Reply:
the function isContactsCommonList is being called inside the render function. So if isContactsCommonList is true, then the component will render nothing, else it will render the common contacts.

Reply:
Thank you for elaborating @RGHuynh , I see the intention. However just calling the function doesn't interrupt the render() and the component is rendered in its fullness. One can see an empty box when displaying a user with contacts but no contacts in common.

Reply:
I see what you're saying. It must of went over my head haha. My intention was to encapsulate the isContactsCommonList and call it within the render, but i can see how it won't work. I'll fix it right now.

Reply:
updated